tests/system/stats/stackdriver/stackdriver_stats_test.py

```diff
@@ -0,0 +1,153 @@
+from opencensus.tags import tag_map as tag_map_module
+from opencensus.tags import tag_value as tag_value_module
+from opencensus.common.transports import sync
+from google.cloud import monitoring_v3
+
+MiB = 1 << 20
+
+PROJECT = os.environ.get('GCLOUD_PROJECT_PYTHON')
+RETRY_WAIT_PERIOD = 10000  # Wait 10 seconds between each retry
+RETRY_MAX_ATTEMPT = 10  # Retry 10 times
+
+class TestBasicStats(unittest.TestCase):
+    
+    @retry(wait_fixed=RETRY_WAIT_PERIOD, stop_max_attempt_number=RETRY_MAX_ATTEMPT)
+    def test_stats_record_sync(self):
+        # We are using sufix in order to prevent cached objects
+        sufix = str(os.getgid())
+
+        tag_key = "SampleKeySyncTest%s" % sufix
+        measure_name = "SampleMeasureNameSyncTest%s" % sufix
+        measure_description = "SampleDescriptionSyncTest%s" % sufix
+        view_name = "SampleViewNameSyncTest%s" % sufix
+        view_description = "SampleViewDescriptionSyncTest%s" % sufix
+
+        FRONTEND_KEY = tag_key_module.TagKey(tag_key)
+        VIDEO_SIZE_MEASURE = measure_module.MeasureInt(
+            measure_name, measure_description, "By")
+        VIDEO_SIZE_VIEW_NAME = view_name
+        VIDEO_SIZE_DISTRIBUTION = aggregation_module.DistributionAggregation(
+                                    [0.0, 16.0 * MiB, 256.0 * MiB])
+        VIDEO_SIZE_VIEW = view_module.View(VIDEO_SIZE_VIEW_NAME,
+                                        view_description,
+                                        [FRONTEND_KEY],
+                                        VIDEO_SIZE_MEASURE,
+                                        VIDEO_SIZE_DISTRIBUTION)
+
+        stats = stats_module.Stats()
+        view_manager = stats.view_manager
+        stats_recorder = stats.stats_recorder
+
+        client = monitoring_v3.MetricServiceClient()
+        exporter = stackdriver.StackdriverStatsExporter(options=stackdriver.Options(project_id=PROJECT),
+                                                        client=client,
+                                                        transport=sync.SyncTransport)
+        view_manager.register_exporter(exporter)
+
+        # Register view.
+        view_manager.register_view(VIDEO_SIZE_VIEW)
+
+        # Sleep for [0, 10] milliseconds to fake work.
+        time.sleep(random.randint(1, 10) / 1000.0)
+
+        # Process video.
+        # Record the processed video size.
+        tag_value = tag_value_module.TagValue("1200")
+        tag_map = tag_map_module.TagMap()
+        tag_map.insert(FRONTEND_KEY, tag_value)
+        measure_map = stats_recorder.new_measurement_map()
+        measure_map.measure_int_put(VIDEO_SIZE_MEASURE, 25 * MiB)
+
+        measure_map.record(tag_map)
+
+
+        # Sleep for [0, 10] milliseconds to fake wait.
+        time.sleep(random.randint(1, 10) / 1000.0)
+
+        name = exporter.client.project_path(PROJECT)
+        list_metrics_descriptors = exporter.client.list_metric_descriptors(name)
```

Comment:
Retry logic should be added here instead of adding to the whole test, we want to wait for the metric creation and we don't want to create the metric again when failing.